Add config parameter to allow unsafe access for infiniband uverbs.
Subject: infiniband-uverbs: Introduce config parameter to allow unsafe access.

Detailed description of changes follows

Security contexts changes opening file desciptior which is not allowed and
hence infiniband uverbs write access fails. This patch allows to make
this accces a configurable parameter.

Signed-off-by: Sameer Agrawal <sagrawa@vyatta.att-mail.com
--- a/drivers/infiniband/Kconfig
+++ b/drivers/infiniband/Kconfig
@@ -88,6 +88,13 @@ config INFINIBAND_ADDR_TRANS_CONFIGFS
 	  This allows the user to config the default GID type that the CM
 	  uses for each device, when initiaing new connections.
 
+config INFINIBAND_UVERBS_ALLOW_UNSAFE_ACCESS
+        bool
+        depends on INFINIBAND
+        default y
+        ---help---
+          Support to allow uverbs unsafe access.
+
 source "drivers/infiniband/hw/mthca/Kconfig"
 source "drivers/infiniband/hw/qib/Kconfig"
 source "drivers/infiniband/hw/cxgb3/Kconfig"
--- a/include/rdma/ib.h
+++ b/include/rdma/ib.h
@@ -100,7 +100,11 @@ struct sockaddr_ib {
  */
 static inline bool ib_safe_file_access(struct file *filp)
 {
+#if IS_ENABLED(CONFIG_INFINIBAND_UVERBS_ALLOW_UNSAFE_ACCESS)
+	return true;
+#else	
 	return filp->f_cred == current_cred() && !uaccess_kernel();
+#endif	
 }
 
 #endif /* _RDMA_IB_H */
